amazon-eks-al2.pkr.hcl

```diff
@@ -0,0 +1,113 @@
+locals {
+  timestamp = regex_replace(timestamp(), "[- TZ:]", "")
+
+  target_ami_name = "${var.ami_name_prefix}-${var.eks_version}-${local.timestamp}"
+
+  e2_test_subnet_id = data.amazon-parameterstore.subnet_id.value
+}
+
+data "amazon-ami" "this" {
+  filters = {
+    architecture        = var.source_ami_arch
+    name                = "${var.ami_name_prefix}-${var.eks_version}-*"
+    root-device-type    = "ebs"
+    state               = "available"
+    virtualization-type = "hvm"
+  }
+
+  most_recent = true
+  owners = [
+    var.source_ami_owner,
+    var.source_ami_owner_govcloud,
+  ]
+  region = var.aws_region
+}
+
+data "amazon-parameterstore" "subnet_id" {
+  name            = "/amazon-eks-custom-amis/tests/subnet-id"
```

Comment:
this value comes from the SSM parameter created in the `tests/vpc` directory to aid in e2e testing